src/GitVersion.Core/VersionCalculation/Abstractions/IBaseVersionCalculator.cs

```diff
@@ -1,6 +1,8 @@
+using GitVersion.Model.Configuration;
+
 namespace GitVersion.VersionCalculation;
 
 public interface IBaseVersionCalculator
 {
-    BaseVersion GetBaseVersion();
+    (BaseVersion, EffectiveBranchConfiguration) GetBaseVersion();
```

Comment:
An alternative to using tuple here would be to create a class holding the `BaseVersion` and `EffectiveBranchConfiguration` properties. I think I prefer that over tuple, tbh. Especially if you can find a good, descriptive name for it. :)

Reply:
I haven't introduce a class because I would like to fuse the BaseVersionCalculator with NextVersionCalculator in the v6 refactoring. But if you disagree than I can do it no problem. Please let me know.